pywbem/_subscription_manager.py

```diff
@@ -1284,16 +1309,87 @@ def _create_filter(self, server_id, source_namespace, query,
                         filter_inst = server.conn.GetInstance(filter_path)
                         self._owned_filters[server_id][i] = filter_inst
                     return filter_inst
-            filter_path = server.conn.CreateInstance(filter_inst)
+
+            filter_path = server.conn.CreateInstance(
+                filter_inst, namespace=server.interop_ns)
             filter_inst = server.conn.GetInstance(filter_path)
             self._owned_filters[server_id].append(filter_inst)
         else:
             # Responsibility to ensure it does not exist yet is with the user
-            filter_path = server.conn.CreateInstance(filter_inst)
+            filter_path = server.conn.CreateInstance(
+                filter_inst, namespace=server.interop_ns)
             filter_inst = server.conn.GetInstance(filter_path)
 
+        # If systemname does not yet exist, create it
+        if server_id not in self._systemnames:
+            self._systemnames[server_id] = filter_inst['SystemName']
+        else:
+            if self._systemnames[server_id] != filter_inst['SystemName']:
+                warnings.warn(
+                    _format("WBEM server may be returning different SystemName "
+                            "property for different operations: received  "
+                            "{0!A} and {0!A}. Owned filters may not be found.",
+                            self._systemnames[server_id],
+                            filter_inst['SystemName']),
+                    UserWarning)
+
         return filter_inst
 
+    def _get_systemname(self, server_id):
+        """
+        Get the SystemName returned for anything returned for this server_id.
+        This method gets the s
+        """
+        if server_id in self._systemnames:
+            return self._systemnames[server_id]
+
+        return None
+
+    def _build_path(self, inst, ns, server_id):
+        """
+        Common method to build a CIM_InstanceName for either filter or
+        destination. Used to build the paths for both destinations and filters.
+        Keybindings will be included only if a system name has already
+        been retrieved from an earlier add_filter or add_destination.
+
```

Comment:
It would be worthwhile to point out that this common method relies on the fact that CIM_IndicationFilter and CIM_ListenerDestination happen to have the same key properties.

Reply:
DONE